Engine/DynamicMemberTypeDataProvider.cs

```diff
@@ -467,6 +467,7 @@ internal static ParameterMemberData ParameterizeMember(object target, IMemberDat
             if (member == null) throw new ArgumentNullException(nameof(member));
             if (source == null) throw new ArgumentNullException(nameof(source));
             if (name == null) throw new ArgumentNullException(nameof(name));
+            if (source == target) throw new InvalidOperationException("Source and target cannot be the same object.");
```

Comment:
This fix looks unrelated to this issue. Did you encounter this scenario while working on this?

Reply:
This happened when I tried to implement the test. If the check is not done a stack overflow exception will occur.